### PR TITLE
Stickers market UI

### PR DESCRIPTION
**Pull request by flexsurfer**
Implementation of https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat 

https://drive.google.com/file/d/15ffNyWFNBfOlftk5yW0RxmkyP7ZzTDjR/view

<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat">Figma</a></strong></div><div>Created with Figma</div></blockquote>
<blockquote><div>Google Drive</div><div><strong><a href="https://drive.google.com/file/d/15ffNyWFNBfOlftk5yW0RxmkyP7ZzTDjR/view?usp=embed_facebook">Jan-07-2019 13-23-09.mp4</a></strong></div></blockquote>

### CODE REVIEW

**Comment by jeluard** [comment]
@flexsurfer Nice job!
Do we want to keep the PR open? Or just keep a shared branch for now?

**Comment by flexsurfer** [comment]
i think it's fine to keep it open, I believe we'll continue work soon :)

**Comment by errorists** [comment]
dude, you're on fire 🔥 🚒  one thing which isn't obvious from the designs - sorry - it would be awesome if the sticker selection menu would get its position / height from the system keyboard so when you're switching between the input and stickers there's no 'jump'. like so
screenshot 2019-01-07 at 15 20 08

**Comment by flexsurfer** [comment]
hey dude, this is an early prototype, you shouldn't have seen this :)  thanks for the comment, makes sense, I'll try to address it later :)

**Review by goranjovic — APPROVED** [review]

File: src/status_im/chat/models/input.cljs [hunk 1]
```diff
@@ -147,6 +147,15 @@
...
                                                             :text    message-text})})))
 
+(fx/defn send-sticker-fx
+  [{:keys [db] :as cofx} uri current-chat-id]
+  (when-not (string/blank? uri)
+    (chat.message/send-message cofx {:chat-id      current-chat-id
+                                     :content-type constants/content-type-sticker
+                                     :content      (cond-> {:chat-id current-chat-id
+                                                            :uri     uri
+                                                            :text    "Update to latest version to see a nice sticker here!"})})))
```
Comment (jeluard) [inline src/status_im/chat/models/input.cljs:157]: i18n?

File: src/status_im/events.cljs [hunk 2]
```diff
@@ -775,6 +777,14 @@
  (fn [{{:keys [current-chat-id]} :db :as cofx} [_ message-text]]
    (chat.input/send-plain-text-message-fx cofx message-text current-chat-id)))
 
+(handlers/register-handler-fx
+ :chat/send-sticker
+ (fn [{{:keys [current-chat-id] :account/keys [account]} :db :as cofx} [_ {:keys [uri]}]]
+   (fx/merge
+    cofx
+    (accounts/update-recent-stickers (conj (remove #(= uri %) (:recent-stickers account)) uri))
```
Comment (jeluard) [inline src/status_im/events.cljs:785]: Add some comment?

**Review by jeluard — APPROVED** [review]

**Comment by asemiankevich** [comment]
image
 
looks good to me for now, assume it is initial state :) Thanks @flexsurfer great job

**Comment by errorists** [comment]
yeah, great job @flexsurfer 💪 👏 

i did a design review and here are the issues I found

- [ ] the icon for Commands is black, should be Grey
- [ ] a beautiful thing that would make me cry is if the menu wouldn't 'jump' on opening but animate instead. But this is a larger problem as the text input also has this wonky issue where only the keyboard animates smoothly and the input just teleports into position.
- [ ] like [I've mentioned before](https://github.com/status-im/status-react/pull/7210#issuecomment-451949739), the Sticker Menu should grab its height from the system keyboard
- [ ] The 'Get stickers' button doesn't have touch feedback, it should be a Touchable with Opacity
- [ ] The Sticker Market toolbar/top nav is too high, it looks like it's 108px, should be a 100
- [ ] Sticker Market, tapping on the smaller sticker thumbnail, title or artist name should also lead you to sticker details, the only area that's outside of that is the install button.
- [ ] Sticker Pack Details, the thumbnail should be a 64 by 64
- [ ] Sticker Pack Details, the thumbnail, title and subtitle are too low, I guess the toolbar here is again 8px too high pushing them down.
- [ ] Sticker Pack Details, the entire top toolbar is too high, should be a 188 instead of 202px
- [ ] the underline under selected sticker pack should have a 1px border radius. We're designers, we round everything ;) 
- [ ] a non-issue yet since we won't have many packs at launch but the sticker pack selection menu should be horizontally scrollable
- [ ] the sticker pack menu, the one where you can select the sticker to send, should be [horizontally page'able](https://facebook.github.io/react-native/docs/scrollview#pagingenabled) so you can swipe between sticker packs.
- [ ] tapping on a sent sticker should display the sticker details view.

**Comment by flexsurfer** [comment]
thanks @errorists i filed the issues https://github.com/status-im/status-react/issues/7374

**Pull request merged**